src/main/java/seedu/meditracker/command/AddCommand.java

```diff
@@ -0,0 +1,60 @@
+package seedu.meditracker.command;
+
+import seedu.meditracker.exception.MediTrackerException;
+import seedu.meditracker.medication.Medication;
+import seedu.meditracker.medication.MedicationList;
+import seedu.meditracker.ui.Ui;
+
+/**
+ * The AddCommand class represents a command to add a new medication.
+ * It extends the Command class.
+ */
+public class AddCommand extends Command {
+
+    private static final int MED_NAME = 1;
+    private static final int MED_QTY = 3;
+    private static final int MED_DOSAGE = 5;
+    private static final int MED_EXP = 7;
+    private static final int MED_FREQ = 9;
+    private static final int MED_RMK = 11;
+
+    private String medicationName;
+    private String medicineQuantity;
+    private String medicineDosage;
+    private String expiryDate;
+    private String intakeFreq;
+    private String remarks;
+
+    /**
+     * Constructs an AddCommand object with the specified arguments.
+     * @param arguments The arguments containing medication information to be parsed.
+     */
+    public AddCommand(String arguments) throws MediTrackerException {
+        try {
+            String[] medicineInformation = arguments.split(" ");
+            medicationName = medicineInformation[MED_NAME];
+            medicineQuantity = medicineInformation[MED_QTY];
+            medicineDosage = medicineInformation[MED_DOSAGE];
+            expiryDate = medicineInformation[MED_EXP];
+            intakeFreq = medicineInformation[MED_FREQ];
+            remarks = medicineInformation[MED_RMK];
+        } catch (Exception e) {
+            throw new MediTrackerException();
+        }
```

Comment:
Good job on handling the exception, but do note that you should avoid catching generic exception and explicitly specify the exact exception type that will be caught. Also, it would be better practice to scope the try catch to the lines that will throw the exception.

```suggestion
        String[] medicineInformation = arguments.split(" ");
        try {
            medicationName = medicineInformation[MED_NAME];
            medicineQuantity = medicineInformation[MED_QTY];
            medicineDosage = medicineInformation[MED_DOSAGE];
            expiryDate = medicineInformation[MED_EXP];
            intakeFreq = medicineInformation[MED_FREQ];
            remarks = medicineInformation[MED_RMK];
        } catch (ArrayIndexOutOfBoundsException e) {
            throw new MediTrackerException();
        }
```